### ISSUE TITLE
Code review

### ISSUE THREAD

**Issue by taneltm** [flamableassassin/discord.js-bot-base#1]
**-[ package.json ]-----**
* `//1` doesn't do anything, should be removed.
* `"name": "name here"` - maybe have the project name there?
* `dependencies` is empty, but the project depends on discord.js, express, discord-bot-analyticsm and dblapi.js.
  * `npm install discord.js`
  * `npm install express`
  * `npm install discord-bot-analytics`
  * `npm install dblapi.js`
* Repository url should be `https://github.com/flamableassassin/discord.js-bot-base`
* Keywords
  * Now that you have the project in github, it's no longer related to glitch.
  * You are missing the most important keywords: bot, discord, discord.js

**-[ .gitignore ]-----**
* File missing - you should add it and make git ignore the node_modules directory and log files.
* You can use this as basis https://github.com/github/gitignore/blob/master/Node.gitignore
  * The file name should be `.gitignore` not `Node.gitignore` tho.

**-[ .editorconfig ]-----**
* File missing - you should add it. It contains basic configurations for IDEs and code editors.
* You can use this as basis https://github.com/airbnb/javascript/blob/master/.editorconfig
* You can read more about it here https://editorconfig.org/
* Make sure your code editor enforces the rules after adding the file, for example make an indentation with 3 spaces and your code editor should complain about it.
  * If your code editor doesn't complain, you'll need to install a plugin for your editor https://editorconfig.org/#download
* If you want to further improve code consistency, use https://eslint.org/
  * Configure your editor to use it
  * Setup a git commit hook with Husky (https://www.npmjs.com/package/husky) so that the code will be linted automatically before a git commit.

**-[ bot.js ]-----**
* Group your imports (in this case requires)
  * Start your js file with importing everything you need to import. If you have imports in the middle of your code, it's harder to clean up your unused imports later.
  * Have an empty line after declaring other variables.
  * The `src/index.js` file would be an exception to this rule - you don't need to do that there.
* console.log is fine, but if you need more info, for example what's going on with express, check out morgan
  * `npm install morgan`
  * https://www.npmjs.com/package/morgan#expressconnect

**-[ src/commands/base.js ]-----**
* There should be a space before and after an `=` sign in an assignment statememnt (line 1)
* Mixed tabs and spaces - `.editorconfig` should help with that
* No space after `"usage":` on line 6.
* When you have a comment after a code line, there should be a space before `//`.
* There should be a space between `//` and the comment text.

**-[ src/loaders/command-loader.js ]-----**
* There should be a space before and after an `=` sign in an assignment statememnt (line 1)
* `keys` is not defined on line 9.
* It took me a moment to understand how the loading is supposed to work. There should be example commands to make it easier to understand for others.
* Line 3, `client.commands = new Discord.Collection();` - you are modifying the Discord.js Client object.
  * If Discord.js adds a `commands` attribute in a future version, it will break your bot.

**-[ src/loaders/event-loader.js ]-----**
* There should be a space before and after an `=` sign in an assignment statememnt (line 1)
* Should this file have a commment "Not implemented yet!" ?

**-[ src/loaders/index.js ]-----**
* There should be a space before and after an `=` sign in an assignment statememnt (line 1)
* Requires should be outside the exported function, otherwise you're re-requiring every time you execute the exported function.
* Separate imports (requires) from the rest of the code with an empty line.

**-[ src/loaders/module.js ]-----**
* Unnecessary space between return statement and semicolon.
* There should be a space between `//` and the comment text.

**-[ src/watchers/command.js ]-----**
* There should be a space before and after an `=` sign in an assignment statememnt (line 1)
* Requires should be outside the exported function, otherwise you're re-requiring every time you execute the exported function.
* Separate imports (requires) from the rest of the code with an empty line.
* Mixed tabs and spaces - `.editorconfig` should help with that.
** It's really hard to follow the code with inconsitent indentation.
* You are lacking spaces around operators in several places. ESLint would help you fix those.
* The exported function has too much complexity, you could create separate named functions for all the main IF cases.

**-[ src/watchers/index.js ]-----**
* There should be a space before and after an `=` sign in an assignment statememnt (line 1)
* Requires should be outside the exported function, otherwise you're re-requiring every time you execute the exported function.
* Separate imports (requires) from the rest of the code with an empty line.

**-[ src/watchers/other.js ]-----**
* There should be a space before and after an `=` sign in an assignment statememnt (line 1)
* If you start using `morgan`, this would be the place to use it for discord.js
* Maybe call this file `logging.js`?

**-[ src/watchers/ready.js ]-----**
* There should be a space before and after an `=` sign in an assignment statememnt (line 1)
* Requires should be outside the exported function, otherwise you're re-requiring every time you execute the exported function.
* Separate imports (requires) from the rest of the code with an empty line.
* Line 7, `name: 'Made by flammableassassin'` - this should be configurable.
* Line 14 is too long. Decide on a max line length for the project, usually 80 or 120 characters and try to stick to it.
* RichEmbed setColor hex code should be prefixed with `#`
  * Example: https://discordjs.guide/popular-topics/embeds.html#embed-preview
* Line 17, `YOR_LOG_CHANNEL_HERE` - YOR?

**-[ src/watchers/serverjoin.js ]-----**
* Some code duplication with `src/watchers/ready.js`, consider creating a shared function.

**-[ src/watchers/userjoin.js ]-----**
* There should be a space before and after an `=` sign in an assignment statememnt (line 1)
* Inconsistent indentation - single space this time
* Spaces between arguments
  * This `client.on("guildMemberAdd",member=>{`
  * Should be `client.on("guildMemberAdd", member => {`
* It might also be a good idea to surround the arrow function's arguments with brackets, this way it's easier to read - you'll instantly know that `member` belongs to the arrow function and isn't the `client.on` function's argument.
  * `client.on("guildMemberAdd", (member) => {`

**Comment by flamableassassin** [comment]
No offense but thanks for the help but its not really the help im looking for.
